core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -94,6 +94,11 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
   // contains a map from hostname to a list of input format splits on the host.
   private[spark] var preferredNodeLocationData: Map[String, Set[SplitInfo]] = Map()
 
+  // This is used for Spark Streaming to check whether driver host and port are set by user,
+  // if these two configurations are set by user, so the recovery mechanism should not remove this.
+  private[spark] val isDriverHostSetByUser = config.contains("spark.driver.host")
```

Comment:
I'm not sure how to track this in `Checkpoint`, since `SparkEnv` will reset this configuration if user not set it, so in the `Checkpoint` how to differentiate whether this is set by user or `SparkEnv`?